contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1628,8 +1649,12 @@ check_alter_server_stmt(GrantRoleStmt *stmt)
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("%s is not a login", grantee_name)));
 
-	/* only sysadmin role is assumed below */
-	if (!has_privs_of_role(GetSessionUserId(), sysadmin))
+	/* 
+	 * check if it has sysadmin privileges or
+	 * if server role is securityadmin and it has privileges of securityadmin
+	 */
+	if (!has_privs_of_role(GetSessionUserId(), sysadmin) && ((strcmp(granted_name, "securityadmin") != 0)
+										|| !has_privs_of_role(GetSessionUserId(), securityadmin_oid)))
```

Comment:
What happens when we alter try to alter sysadmin role? For that case as well we should be throwing this kind of error. But from the current implementation, I don't think we will be throwing this error. Can you please validate and modify this check to include sysadmin check as well if required.

Reply:
Yes it does include all the required check.